src/python/zquantum/core/interfaces/optimizer.py

```diff
@@ -105,3 +106,50 @@ def construct_history_info(cost_function, keep_history):
     if keep_history and hasattr(cost_function, "gradient"):
         histories["gradient_history"] = cost_function.gradient.history
     return histories
+
+
+class MetaOptimizer(ABC):
+    def __init__(
+        self,
+        ansatz: Ansatz,
```

Comment:
This probably shouldn't be a part of the interface.
Ansatz could incorporated in `estimation_tasks_factory` / `cost_function_factory` or in `**kwargs`.

Reply:
Probably **kwargs.

Reply:
Or just remove `ansatz` alltogether and leave it up to the implementation of the interface. Similarly to what we do with `Optimizer` interface, where we specify only `recorder` in the base class and then we overwrite `init` in the implementations with other inputs as well.
@alexjuda

Reply:
Talked with @dexter2206 , seems that your changes @laurgao are good, we don't want to do use `kwargs`.

Reply:
@mstechly could you share what's the motivation of specifying `__init__` in the interface? It has a bunch of downsides, like:
- Different implementations might need different parameters in the constructor. By specyfing `__init__` in the interface we forbid any other sets of constructor parameters than the ones here.
- in this init we create object fields. If an object's class does inherit this class, the object would have these fields as well. If the object's class doesn't inherit this one, the object would have a different set of fields. This seems like a too much effects when we just want to specify common interface for the `minimize()` method.